File: sdk/swift/Sources/HiTechCloudSDK/ServiceStatus.swift
import Foundation

/// Service Status: list and manage service statuses
public class ServiceStatusResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/statuses - Returns a list of all statuses
    public func list(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/statuses", params: params)
    }

    /// PUT /api/statuses/{id} - Returns details of status
    public func get(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.put("/api/statuses/{id}", params: params)
    }

}

File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/ServiceStatus.kt
package vn.hitechcloud.sdk

/** Service Status: list and manage service statuses */
class ServiceStatusResource(private val httpClient: HttpClient) {
    /** GET /api/statuses - Returns a list of all statuses */
    fun list(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/statuses", params)
    }

    /** PUT /api/statuses/{id} - Returns details of status */
    fun get(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.put("/api/statuses/{id}", params)
    }

}
